### PR TITLE
Refactor DrinkRecipe

### LINKED ISSUE #45: Add propTypes to drinkRecipe

### LINKED ISSUE #51: Media queries for drinkRecipe card

### PR DESCRIPTION
**Pull request by tarynmartin**
### Type of change made:
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Styling- no new features
- [x] Testing

### Detailed Description / What does this PR do?
- Refactor DrinkRecipe to add key to map function, export mapDispatchToProps & mapStateToProps, added PropTypes, and rename a function to be more accurate
- Add search to render in App to display on DrinkRecipe component
- Complete styling for component to be more responsive - may still need some work, but pretty much there
- Create outline of needed tests - only mapStateToProps is currently passing
- Add dependencies for testing

### Files modified:
- DrinkRecipe.js, DrinkRecipe.css, DrinkRecipe.test.js, App.js, packages.json

### Has This Been Tested?
- [ ] No
- [x] Yes

### Are there specific methods for testing?
- in terminal, run npm test to see the mapStateToProps test passing
- in terminal, run npm start. in browser, search for an ingredient, click on a drink, and resize window to see functionality

### What are the relevant tickets?
resolves #45 
resolves #51 
#63

### Message/Questions for reviewer:
- How do I do the fetch testing? Still really lost on this
- May need some adjustments for styling at full width - block isn't centered on the page?
- Will add an icon for the back button in the future, maybe?

### Checklist:
- [ ] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/DrinkRecipe/DrinkRecipe.css [hunk 1]
```diff
@@ -1,32 +1,148 @@
...
+
+  .drink-title {
+    font-size: 1.2em;
+  }
+
+  .ingredients,
+  .instructions {
+    font-size: 1em;
+  }
+}
```
Comment (relyt4me) [inline src/components/DrinkRecipe/DrinkRecipe.css:148]: Thats some impressive CSS

File: src/components/DrinkRecipe/DrinkRecipe.test.js [hunk 2]
```diff
@@ -1,38 +1,278 @@
...
+  let store;
+
+  beforeEach(() => {
+    store = mockStore({
+      recipe = {}
+    })
+  });
+  it('should create a recipe on page load', () => {
+    fetchDrinkRecipe.mockResolvedValue({
+      "drinks": [
```
Comment (relyt4me) [inline src/components/DrinkRecipe/DrinkRecipe.test.js:28]: Could we possibly cutdown this mock resolve for the test. I am never sure if its better to do more or less here.

**Review by relyt4me — APPROVED** [review]

File: src/components/DrinkRecipe/DrinkRecipe.js [hunk 3]
```diff
@@ -59,47 +59,58 @@ class DrinkRecipe extends Component {
...
             </div>
-          </article>
+          </div>
         </section>
       )
-    } else {
+    } else if (hasErrored !== '') {
       return (
-        <h2>Loading...</h2>
+        <h2 className='error-message'>{hasErrored}</h2>
```
Comment (rwilliams659) [inline src/components/DrinkRecipe/DrinkRecipe.js:67]: Nice adding the specific error msg display here

File: src/components/DrinkRecipe/DrinkRecipe.test.js (hunk 2 above)
Comment (tarynmartin) [reply src/components/DrinkRecipe/DrinkRecipe.test.js:28]: I'm not sure. Rachel was saying that the mock resolve is what we get from the response.json. I can probably shorten some stuff, but I'm not sure about whether I can take out some of those strIngredient and strMeasures without messing up the test?

File: src/components/DrinkRecipe/DrinkRecipe.test.js [hunk 4]
```diff
@@ -1,38 +1,278 @@
...
+import { createStore } from 'redux';
+import rootReducer from '../../reducers/index.js';
+import { Provider } from 'react-redux';
+import  configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { createDrinkRecipe, createError } from '../../actions/index.js'
+jest.mock('../../helpers/apiCalls');
+import { fetchDrinkRecipe } from '../../helpers/apiCalls';
+
+const middleware = [thunk]
```
Comment (rwilliams659) [inline src/components/DrinkRecipe/DrinkRecipe.test.js:15]: Given how Tyler was able to get his Search tests working with the async functions without using thunk, I'm wondering if you can as well in this component especially (I'm thinking in App it's more likely we would need the thunk stuff). I might be wrong, but probably worth trying

**Review by rwilliams659 — APPROVED** [review]
Nice job on the styling! Re the testing, based on how Tyler got his search testing working, I think we might be able to work through DrinkRecipe testing together tmrw (I hope!)

**Pull request merged**